--- arch/arm/mach-omap2/omap3-iommu.c	2009-11-27 15:47:53.000000000 +0530
+++ arch/arm/mach-omap2/omap3-iommu-new.c	2009-11-27 15:44:35.000000000 +0530
@@ -36,13 +36,13 @@
 		.nr_tlb_entries = 8,
 		.clk_name = "cam_ick",
 	},
-#if defined(CONFIG_MPU_BRIDGE_IOMMU)
+/*#if defined(CONFIG_MPU_BRIDGE_IOMMU)*/
 	{
 		.name = "iva2",
 		.nr_tlb_entries = 32,
 		.clk_name = "iva2_ck",
 	},
-#endif
+/*#endif*/
 };
 #define NR_IOMMU_DEVICES ARRAY_SIZE(omap3_iommu_pdata)
 
